src/test/java/com/puppycrawl/tools/checkstyle/checks/coding/MatchXpathCheckTest.java

```diff
@@ -230,4 +233,18 @@ public void testGetRequiredTokens() {
                 .that(matchXpathCheck.getRequiredTokens())
                 .isEmpty();
     }
+
+    @Test
+    public void testCheckWithMatchingMethodName2() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(MatchXpathCheck.class);
+        checkConfig.addProperty("query", "count(*) div 0");
```

Comment:
Please share what blocks you to define Check config in Inputs file, as all other are doing

Reply:
@romani None of them are used to test an exception. I need the exact message to kill the mutation(I have mentioned the reason in description) if I define the config in the input file then I have to use the verifyWithInlineConfigParser and it is not able to test that. currently their if we want to test exception using input then verify method is only option.

Reply:
If you will do CLI execution, do you see this part of message?

Reply:
https://github.com/checkstyle/checkstyle/pull/13242/files#r1242293113

```
kevin@inspiron-15-5510:~/Desktop/check_style$ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN" 
"https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="charset" value="ISO-8859-5" />
    <property name="severity" value="warning" />
    <module name="TreeWalker">
        <module name="MatchXpath">
            <property name="query" value="count(*) div 0" />
        </module>

    </module>

</module>

kevin@inspiron-15-5510:~/Desktop/check_style$ cat Test.java
public class Test {
}

kevin@inspiron-15-5510:~/Desktop/check_style$java -jar /home/kevin/Downloads/checkstyle-10.12.0-all.jar -c config.xml Test.java
Starting audit...
com.puppycrawl.tools.checkstyle.api.CheckstyleException: Exception was thrown while processing Test.java
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:306)
        at com.puppycrawl.tools.checkstyle.Checker.process(Checker.java:223)
        at com.puppycrawl.tools.checkstyle.Main.runCheckstyle(Main.java:415)
        at com.puppycrawl.tools.checkstyle.Main.runCli(Main.java:338)
        at com.puppycrawl.tools.checkstyle.Main.execute(Main.java:195)
        at com.puppycrawl.tools.checkstyle.Main.main(Main.java:130)
Caused by: java.lang.IllegalStateException: Evaluation of Xpath query failed: count(*) div 0
        at com.puppycrawl.tools.checkstyle.checks.coding.MatchXpathCheck.findMatchingNodesByXpathQuery(MatchXpathCheck.java:297)
        at com.puppycrawl.tools.checkstyle.checks.coding.MatchXpathCheck.beginTree(MatchXpathCheck.java:274)
        at com.puppycrawl.tools.checkstyle.TreeWalker.notifyBegin(TreeWalker.java:298)
        at com.puppycrawl.tools.checkstyle.TreeWalker.walk(TreeWalker.java:272)
        at com.puppycrawl.tools.checkstyle.TreeWalker.processFiltered(TreeWalker.java:158)
        at com.puppycrawl.tools.checkstyle.api.AbstractFileSetCheck.process(AbstractFileSetCheck.java:98)
        at com.puppycrawl.tools.checkstyle.Checker.processFile(Checker.java:334)
        at com.puppycrawl.tools.checkstyle.Checker.processFiles(Checker.java:293)
        ... 5 more
Caused by: net.sf.saxon.trans.XPathException: Integer division by zero
        at net.sf.saxon.value.Int64Value.div(Int64Value.java:668)
        at net.sf.saxon.expr.Calculator$IntegerDivInteger.compute(Calculator.java:975)
        at net.sf.saxon.expr.ArithmeticExpression$ArithmeticElaborator.lambda$elaborateForItem$2(ArithmeticExpression.java:494)
        at net.sf.saxon.expr.ArithmeticExpression.evaluateItem(ArithmeticExpression.java:428)
        at net.sf.saxon.expr.ArithmeticExpression.evaluateItem(ArithmeticExpression.java:31)
        at net.sf.saxon.expr.Expression.iterate(Expression.java:887)
        at net.sf.saxon.sxpath.XPathExpression.evaluate(XPathExpression.java:192)
        at com.puppycrawl.tools.checkstyle.checks.coding.MatchXpathCheck.findMatchingNodesByXpathQuery(MatchXpathCheck.java:291)
        ... 12 more
Checkstyle ends with 1 errors.
kevin@inspiron-15-5510:~/Desktop/check_style$ 
```

@romani Yes I can See that message in cli
```
Caused by: java.lang.IllegalStateException: Evaluation of Xpath query failed: count(*) div 0
```

Reply:
> You can create another draft PR with such experiment to let us discuss details on exact code and execution you experience.

I do that

Reply:
So, as you see it, pay attention to "Caused by".
To get to it you need to do exception.getCause() as much as you have such phrase in output.
And you will do assert on it.

Reply:
Done

Reply:
@Kevin222004 please use `verifyWithInlineConfigParser` and set query as part of the input file. It should simplify your test to this:
```
@Test
public void testCheckWithMatchingMethodName2() {
        final CheckstyleException ex = assertThrows(CheckstyleException.class,
                () -> verifyWithInlineConfigParser(getPath("InputMatchXpath5.java")));
        assertThat(ex.getCause().getMessage())
                .isEqualTo("Evaluation of Xpath query failed: count(*) div 0");
}
```

Reply:
Done